src/parse/parser.rs

```diff
@@ -306,30 +306,6 @@ impl<'help, 'app> Parser<'help, 'app> {
     pub(crate) fn _build(&mut self) {
         debug!("Parser::_build");
 
-        //I wonder whether this part is even needed if we insert all Args using make_entries
-        let mut key = Vec::new();
-        let mut counter = 0;
-        for (i, a) in self.app.args.args.iter_mut().enumerate() {
-            if a.index == None && a.is_positional() {
-                counter += 1;
-                a.index = Some(counter);
-                key.push((KeyType::Position(counter), i));
-            }
-
-            // Add args with default requirements
-            if a.is_set(ArgSettings::Required) {
-                debug!("Parser::_build: adding {} to default requires", a.name);
-                let idx = self.required.insert(a.id.clone());
-                // If the arg is required, add all it's requirements to master required list
-                for (_, name) in a.requires.iter().filter(|(val, _)| val.is_none()) {
-                    self.required.insert_child(idx, name.clone());
```

Comment:
I am really suspicious that we are not using this code. I see `self.required` being used to create `usage` below in this file.

Reply:
Even if it does affect something, it does so in a way our tests don't catch and I can't figure. I think I'd be happier to deal with less code when I'll be working on usage strings, especially given that this code should be logically placed somewhere in usage.rs and be executed on post-parse stage, not build stage. I would also prefer this code to not be scattered all over the program.